tasks/internal/update-fbc-catalog-task/update-fbc-catalog-task.yaml

```diff
@@ -64,37 +82,33 @@ spec:
         requests:
           memory: 512Mi
           cpu: '1'
+      volumeMounts:
+        - name: service-account-secret
+          mountPath: /mnt/service-account-secret
+        - name: iib-services-config-vol
+          mountPath: /mnt/iib-services-config
+        - name: iib-overwrite-fromimage-credentials-vol
+          mountPath: /mnt/iib-overwrite-fromimage-credentials
+        - name: publishing-credentials-vol
+          mountPath: /mnt/publishing-credentials
       env:
-        - name: IIB_SERVICE_URL
-          valueFrom:
-            secretKeyRef:
-              name: iib-services-config
-              key: url
-        - name: IIB_OVERWRITE_FROM_INDEX_USERNAME
-          valueFrom:
-            secretKeyRef:
-              name: iib-overwrite-fromimage-credentials
-              key: username
-        - name: IIB_OVERWRITE_FROM_INDEX_TOKEN
-          valueFrom:
-            secretKeyRef:
-              name: iib-overwrite-fromimage-credentials
-              key: token
-        - name: KRB5_CONF_CONTENT
-          valueFrom:
-            secretKeyRef:
-              name: iib-services-config
-              key: krb5.conf
         - name: FBC_FRAGMENTS
           value: $(params.fbcFragments)
-        - name: PUBLISHING_CREDENTIAL
-          valueFrom:
-            secretKeyRef:
-              key: targetIndexCredential
-              name: $(params.publishingCredentials)
-              optional: true
       script: |
         #!/usr/bin/env bash
+        
+        IIB_SERVICE_URL="$(cat /mnt/iib-services-config/url)"
+        KRB5_CONF_CONTENT="$(cat /mnt/iib-services-config/krb5.conf)"
+
+        IIB_OVERWRITE_FROM_INDEX_USERNAME="$(cat /mnt/iib-overwrite-fromimage-credentials/username)"
+        IIB_OVERWRITE_FROM_INDEX_TOKEN="$(cat /mnt/iib-overwrite-fromimage-credentials/token)"
+
+        PUBLISHING_CREDENTIAL_PATH="/mnt/publishing-credentials/targetIndexCredential"
+        if [ -f "${PUBLISHING_CREDENTIAL_PATH}" ] && [ -s "${PUBLISHING_CREDENTIAL_PATH}" ]; then
+          PUBLISHING_CREDENTIAL="$(cat /mnt/publishing-credentials/targetIndexCredential)"
```

Comment:
Since we already created the PUBLISHING_CREDENTIAL_PATH variable, it would make sense to use it here as well, right?

Reply:
Updated